app/models/cloud_network.rb

```diff
@@ -47,6 +47,68 @@ def self.class_by_ems(ext_management_system, external = false)
     end
   end
 
+  def ip_address_total_count
```

Comment:
hm, as I am looking at it, this probably needs to go to OpenStack STI subclass, there are OpenStack specific things

or, we separate base class/openstack class and add checks. e.g. allocation_pools must be defined and ip_address_used_count_live is doing OpenStack specific query.